### PR TITLE
[Keyboard] add 10k

### PR DESCRIPTION
**Pull request by nab-os**
Added a simple split 10 keys keyboard for chord testing

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/10k/config.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#pragma once
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x6869 // nabos
+#define PRODUCT_ID      0x0001 // 10k
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    nabos
+#define PRODUCT         10k (1x5)
+#define DESCRIPTION     A simple chord keyboard
```
Comment (drashna) [inline keyboards/handwired/10k/config.h:26]:
```suggestion
```

File: keyboards/handwired/10k/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline keyboards/handwired/10k/keymaps/default/keymap.c:19]:
```suggestion
```

File: keyboards/handwired/10k/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline keyboards/handwired/10k/keymaps/via/keymap.c:19]:
```suggestion
```

File: keyboards/handwired/10k/rules.mk [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+TERMINAL_ENABLE = no
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
+
+SPLIT_KEYBOARD = yes
+LEADER_ENABLE = no
+LTO_ENABLE = yes
+
+COMBOS_ENABLE = no
```
Comment (drashna) [inline keyboards/handwired/10k/rules.mk:29]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = no        # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
AUDIO_ENABLE = no           # Audio output
RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight.

SPLIT_KEYBOARD = yes
LTO_ENABLE = yes
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by nab-os** [comment]
Changes applied :+1:

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/10k/readme.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# 10k
+
+*Simple 10 keys chord keyboard*
+
+* Keyboard Maintainer: [Nabos](https://github.com/nab-os)
+* Hardware Supported: *Tested with Elite-C V4 and ProMicro*
```
Comment (fauxpark) [inline keyboards/handwired/10k/readme.md:6]:
```suggestion
Simple 10 keys chord keyboard

* Keyboard Maintainer: [Nabos](https://github.com/nab-os)
* Hardware Supported: Tested with Elite-C V4 and ProMicro
```

File: keyboards/handwired/10k/info.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{
+  "keyboard_name": "10k",
+  "maintainer": "Nabos",
+  "url": "",
+  "matrix_pins": {
+    "cols": ["C6", "D7", "E6", "B4", "B5"],
+    "rows": ["B6"]
+  },
+  "usb": {
+    "device_ver": "0x0001",
```
Comment (fauxpark) [inline keyboards/handwired/10k/info.json:10]:
```suggestion
    "device_version": "0.0.1",
```

File: keyboards/handwired/10k/config.h [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x6869 // nabos
+#define PRODUCT_ID      0x0001 // 10k
+#define DEVICE_VER      0x0001
```
Comment (fauxpark) [inline keyboards/handwired/10k/config.h:23]:
```suggestion
```

File: keyboards/handwired/10k/config.h [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+#define PRODUCT         10k (1x5)
+
+/* key matrix size */
+// Rows are doubled-up
+#define MATRIX_ROWS 2
+#define MATRIX_COLS 5
+
+// wiring of each half
+#define MATRIX_COL_PINS { C6, D7, E6, B4, B5 }
+#define MATRIX_ROW_PINS { B6 }
```
Comment (fauxpark) [inline keyboards/handwired/10k/config.h:34]:
```suggestion
```

File: keyboards/handwired/10k/10k.h [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_1x5(\
```
Comment (fauxpark) [inline keyboards/handwired/10k/10k.h:21]:
```suggestion
#define LAYOUT_split_1x5(\
```

File: keyboards/handwired/10k/info.json [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    "pid": "0x0001",
+    "vid": "0x6869"
+  },
+  "layouts": {
+    "LAYOUT": {
```
Comment (fauxpark) [inline keyboards/handwired/10k/info.json:15]:
```suggestion
    "LAYOUT_split_1x5": {
```

File: keyboards/handwired/10k/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+enum layer_names {
+  _QWERTY
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_QWERTY] = LAYOUT_1x5(
```
Comment (fauxpark) [inline keyboards/handwired/10k/keymaps/default/keymap.c:24]:
```suggestion
  [_QWERTY] = LAYOUT_split_1x5(
```

File: keyboards/handwired/10k/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+enum layer_names {
+  _QWERTY
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_QWERTY] = LAYOUT_1x5(
```
Comment (fauxpark) [inline keyboards/handwired/10k/keymaps/via/keymap.c:24]:
```suggestion
  [_QWERTY] = LAYOUT_split_1x5(
```

File: keyboards/handwired/10k/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+     KC_A , KC_S  ,  KC_D  , KC_F  , KC_SPACE  ,         KC_SPACE     , KC_J  , KC_K  , KC_L , KC_SEMICOLON
+  ),
```
Comment (fauxpark) [inline keyboards/handwired/10k/keymaps/via/keymap.c:26]: Need three more empty layers here.

File: keyboards/handwired/10k/rules.mk [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = no        # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+AUDIO_ENABLE = no           # Audio output
+RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight.
+
```
Comment (fauxpark) [inline keyboards/handwired/10k/rules.mk:19]:
```suggestion
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = no        # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/10k/info.json (hunk 6 above)
Comment (nab-os) [reply keyboards/handwired/10k/info.json]: I'm not sure about this suggestion.
When it try it, I get this error:
`usb: Additional properties are not allowed ('device_version' was unexpected)`
And I cannot find it in other keyboards, all I find is things like this:
https://github.com/qmk/qmk_firmware/blob/9dea6f772077dc5c09daf40378e45884d29ab2e2/keyboards/clueboard/60/info.json#L29
But in the documentation it clearly noted "device_version":
https://github.com/qmk/qmk_firmware/blob/master/docs/reference_info_json.md#usb

What do you suggest ?

**Comment by nab-os** [comment]
I tried to move most of what I could into info.json, as I understand this is a modern way of configuring a keyboard.
I hope that I got it the right way.

**Comment by nab-os** [comment]
Also, I tried to move the matrix into info.json, but as this is a split, rows are doubled and I was not sure on how to reflect that in the config.
Here is my attempt:
https://github.com/nab-os/qmk_firmware/blob/10k/keyboards/handwired/10k/info.json#L8
```
"matrix_pins": {
  "cols": ["C6", "D7", "E6", "B4", "B5"],
  "rows": ["B6", "B6"]
},
```
Because B6 will be used on both boards for this rows as they are splitted

**Comment by nab-os** [comment]
I could not find an already existing split keyboard with only a info.json matrix definition.

File: keyboards/handwired/10k/info.json (hunk 6 above)
Comment (fauxpark) [reply keyboards/handwired/10k/info.json]: image

File: keyboards/handwired/10k/info.json [hunk 15]
```diff
@@ -0,0 +1,45 @@
+{
+  "keyboard_name": "10k",
+  "maintainer": "Nabos",
+  "debounce": 5,
+  "diode_direction": "COL2ROW",
+  "matrix_pins": {
+    "cols": ["C6", "D7", "E6", "B4", "B5"],
+    "rows": ["B6", "B6"]
```
Comment (fauxpark) [inline keyboards/handwired/10k/info.json:8]:
```suggestion
    "rows": ["B6"]
```

File: keyboards/handwired/10k/info.json [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+    "console": false,
+    "extrakey": false,
+    "mousekey": false,
+    "nkro": false,
+    "rgblight": false
+  },
+  "usb": {
+    "vid": "0x6869",
+    "pid": "0x0001",
+    "device_ver": "0x0001"
```
Comment (fauxpark) [inline keyboards/handwired/10k/info.json:24]:
```suggestion
    "device_version": "0.0.1"
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by fauxpark** [comment]
> Also, I tried to move the matrix into info.json, but as this is a split, rows are doubled and I was not sure on how to reflect that in the config.

The docs are not up to date, you will need to read the keyboard schema.

File: keyboards/handwired/10k/info.json (hunk 6 above)
Comment (nab-os) [reply keyboards/handwired/10k/info.json]: I'm truly sorry. I don't know how I could miss that.

**Comment by nab-os** [comment]
My git comments are not good, I'm ok with squashing to make it cleaner

**Comment by nab-os** [comment]
Changes applied :+1:

**Comment by nab-os** [comment]
Is there anything that I should do before it can be merged ?
@fauxpark @drashna

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/10k/config.h [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define PRODUCT         10k (1x5)
+
```
Comment (fauxpark) [inline keyboards/handwired/10k/config.h:22]:
```suggestion
```

File: keyboards/handwired/10k/info.json [hunk 18]
```diff
@@ -0,0 +1,45 @@
+{
+  "keyboard_name": "10k",
+  "maintainer": "Nabos",
```
Comment (fauxpark) [inline keyboards/handwired/10k/info.json:3]:
```suggestion
  "maintainer": "Nabos",
  "manufacturer": "Nabos",
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by nab-os** [comment]
Changes applied :+1:

**Comment by nab-os** [comment]
Is there anything else that I can do ?
@fauxpark @drashna

**Review by fauxpark — APPROVED** [review]

**Comment by nab-os** [comment]
I'm not sure if we are waiting for something on my side or not.
I'm not authorized to merge even with the approval.
Let me know if I did not understand something.

**Pull request merged**